src/desktop/apps/personalize/index.ts

```diff
@@ -0,0 +1,59 @@
+import express, { NextFunction } from 'express'
+import { renderLayout } from '@artsy/stitch'
+import { App } from 'desktop/apps/personalize/components/App'
+import { attachControllers, Controller, Get, Middleware } from "@decorators/express"
+
+export const app = express()
+app.set('views', __dirname)
+app.set('view engine', 'jade')
+
+class EnsureLoggedInUserMiddleware implements Middleware {
+  public use(_, response, next: NextFunction): void {
+    if (!response.locals.sd.CURRENT_USER) {
+      return response.redirect('/personalize')
+    }
+
+    next()
+  }
+}
+
+@Controller('/')
+class PersonalizeController {
+  @Get('/personalize')
```

Comment:
I think it's a good idea, it's similar to how [flash](http://flask.pocoo.org/docs/0.12/patterns/viewdecorators/) annotate functions in python. Like the following
```python
@app.route('/')
def index():
    return render_template('index.html', value=42)
```
I think there are real benefits to colocating route definitions with their corresponding handlers. For one, because the handlers are methods in the same class, it encourages make use of shared session objects when appropriate such as caching data across requests.

Reply:
I'm a big fan of decorators, but I'm not sure what they give us over what we have now except for risk around devs abusing 'this' -- meaning, the only reason there's a 'routes.js' file is because it's a pattern that has been around for a while; why not ditch it and just put everything in index and avoid all the abstractions?